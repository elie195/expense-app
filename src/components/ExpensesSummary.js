import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
  const plural = expenseCount === 1 ? '' : 's';
  const formattedTotal = numeral(expensesTotal / 100).format('$0,0.00');
  return (
    <div>
      <p>
        Viewing {expenseCount} expense
        {plural} totalling {formattedTotal}
      </p>
    </div>
  );
};

const mapStateToProps = state => ({
  expenseCount: selectExpenses(state.expenses, state.filters).length,
  expensesTotal: selectExpensesTotal(state.expenses)
});

export default connect(mapStateToProps)(ExpensesSummary);
