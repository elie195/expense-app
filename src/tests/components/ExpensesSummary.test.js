import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

// let expenseCount, expensesTotal;

// beforeEach(() => {
//   expenseCount = jest.fn();
//   expensesTotal = jest.fn();
//   wrapper = shallow(<ExpensesSummary expenseCount={expenseCount} expensesTotal={expensesTotal} />);
// });

test('should render ExpensesSummary with no expenses', () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={0} expensesTotal={0} />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with expenses', () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={3} expensesTotal={114195} />
  );
  expect(wrapper).toMatchSnapshot();
});
