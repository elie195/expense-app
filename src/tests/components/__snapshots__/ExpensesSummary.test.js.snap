// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with expenses 1`] = `
<div>
  <p>
    Viewing 
    3
     expense
    s
     totalling 
    $1,141.95
  </p>
</div>
`;

exports[`should render ExpensesSummary with no expenses 1`] = `
<div>
  <p>
    Viewing 
    0
     expense
    s
     totalling 
    $0.00
  </p>
</div>
`;
